concepts/headers-includes/about.md

```diff
@@ -0,0 +1,92 @@
+# About
+
+C requires that all functions are declared _before_ they are defined.
+This can be done at the top of the `*.c` file containing the definition, but there are advantages in splitting declarations into a separate header file with a `.h` extension.
+
+You can think of the header file as an API.
```

Comment:
I'd prefer "interface" instead of "API" because I believe it's easier to understand for people who don't know what an "API" is. But that's not a strong preference.